datafusion/expr/src/expr.rs

```diff
@@ -962,18 +942,31 @@ fn create_name(e: &Expr, input_schema: &DFSchema) -> Result<String> {
 }
 
 /// Create a comma separated list of names from a list of expressions
-fn create_names(exprs: &[Expr], input_schema: &DFSchema) -> Result<String> {
+fn create_names(exprs: &[Expr]) -> Result<String> {
     Ok(exprs
         .iter()
-        .map(|e| create_name(e, input_schema))
+        .map(|e| create_name(e))
         .collect::<Result<Vec<String>>>()?
         .join(", "))
 }
 
 #[cfg(test)]
 mod test {
     use crate::expr_fn::col;
-    use crate::lit;
+    use crate::{case, lit};
+    use datafusion_common::{Result, ScalarValue};
+
+    #[test]
+    fn format_case_when() -> Result<()> {
```

Comment:
I added this test just to demonstrate the difference between `Expr` `Display`, `Debug`, and `name()`. I plan on adding more tests in future PRs to cover other expressions.